LLama/LLamaWeights.cs

```diff
@@ -64,28 +73,28 @@ public sealed class LLamaWeights
         /// All metadata keys in this model
         /// </summary>
         public IReadOnlyDictionary<string, string> Metadata { get; set; }
+        #endregion
 
-        private LLamaWeights(SafeLlamaModelHandle weights)
+        private LLamaWeights(SafeLlamaModelHandle handle)
         {
-            NativeHandle = weights;
-            Metadata = weights.ReadMetadata();
+            NativeHandle = handle;
+            Metadata = handle.ReadMetadata();
+
+            // Increment the model reference count while this weight exists.
+            // DangerousAddRef throws if it fails, so there is no need to check "success"
+            var success = false;
+            NativeHandle.DangerousAddRef(ref success);
         }
 
+        #region Load
         /// <summary>
-        /// Create from a "shared" handle
+        /// Create from a "shared" handle. The `SafeLlamaModelHandle` will not be disposed and the model will not be unloaded until <b>all</b> such handles have been disposed.
         /// </summary>
         /// <param name="handle"></param>
         /// <returns></returns>
```

Comment:
Can this be made into a `Clone()` method (not sure about that name, open to suggestions). Basically an instance method on `LLamaWeights` which creates a new `LLamaWeights` that shares the same `NativeHandle` and `Metadata` fields. 

Alternatively, it could just be removed altogether, since it's now just a call to the constructor. However adding a clone method would be nice since it skips reading the metadata again.

Reply:
created clone method